### PR TITLE
refactor(ui): split some xhr calls

### PR DESCRIPTION
**Pull request by bnjjj**

### CODE REVIEW

File: engine/api/application_pipeline.go [hunk 1]
```diff
@@ -237,12 +241,8 @@ func (api *API) removePipelineFromApplicationHandler() Handler {
 			return sdk.WrapError(err, "removePipelineFromApplicationHandler> Cannot update application last modified date")
 		}
 
-		if err := tx.Commit(); err != nil {
-			return sdk.WrapError(err, "removePipelineFromApplicationHandler> Cannot commit tx")
-		}
-
 		var errW error
-		a.Workflows, errW = workflowv0.LoadCDTree(api.mustDB(), api.Cache, key, a.Name, getUser(ctx), "", "", 0)
+		a.Workflows, errW = workflowv0.LoadCDTree(tx, api.Cache, key, a.Name, getUser(ctx), "", "", 0)
```
Comment (sguiheux) [inline engine/api/application_pipeline.go]: move below, without tx

File: engine/api/environment.go [hunk 2]
```diff
@@ -248,7 +248,8 @@ func (api *API) updateEnvironmentsHandler() Handler {
 			}
 		}
 
-		if err := project.UpdateLastModified(tx, api.Cache, getUser(ctx), proj); err != nil {
+		fmt.Println("iciiii", sdk.ProjectEnvironmentLastModificationType)
```
Comment (sguiheux) [inline engine/api/environment.go]: laaaaaa

File: engine/api/project/dao.go [hunk 3]
```diff
@@ -202,6 +194,20 @@ func UpdateLastModified(db gorp.SqlExecutor, store cache.Store, u *sdk.User, pro
 	return nil
 }
 
+func SendLastModifiedEvent(store cache.Store, user *sdk.User, key, updateType string) error {
```
Comment (sguiheux) [inline engine/api/project/dao.go]: go doc

File: engine/api/workflow.go [hunk 4]
```diff
@@ -199,7 +199,7 @@ func (api *API) putWorkflowHandler() Handler {
 			return sdk.WrapError(err, "putWorkflowHandler> Cannot update workflow")
 		}
 
-		if err := project.UpdateLastModified(tx, api.Cache, getUser(ctx), p); err != nil {
+		if err := project.UpdateLastModified(tx, api.Cache, getUser(ctx), p, sdk.ProjectWorkflowLastModificationType); err != nil {
```
Comment (sguiheux) [inline engine/api/workflow.go]: only if the name changed

File: ui/src/app/service/project/project.resolver.ts [hunk 5]
```diff
@@ -2,15 +2,45 @@ import {Injectable} from '@angular/core';
...
 import {RouterService} from '../router/router.service';
 
 @Injectable()
 export class ProjectResolver implements Resolve<Project> {
 
     resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<any>|Promise<any>|any {
         let params = this.routerService.getRouteSnapshotParams({}, state.root);
-        return this.projectStore.getProjectResolver(params['key']);
+        let opts = [
+          new LoadOpts('withVariables', 'variables', this.projectStore.getProjectVariablesResolver),
```
Comment (sguiheux) [inline ui/src/app/service/project/project.resolver.ts]: remove last arguments

File: ui/src/app/service/project/project.resolver.ts [hunk 6]
```diff
@@ -2,15 +2,45 @@ import {Injectable} from '@angular/core';
...
+    constructor(private projectStore: ProjectStore, private routerService: RouterService) {}
+}
+
+@Injectable()
+export class ProjectForApplicationResolver implements Resolve<Project> {
+
+    resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<any>|Promise<any>|any {
+        let params = this.routerService.getRouteSnapshotParams({}, state.root);
+        let opts = [
+          new LoadOpts('withPipelines', 'pipelines', this.projectStore.getProjectPipelinesResolver),
```
Comment (sguiheux) [inline ui/src/app/service/project/project.resolver.ts]: null 3rd arguments

File: sdk/project.go [hunk 7]
```diff
@@ -60,9 +62,14 @@ type LastModification struct {
 }
 
 const (
```
Comment (fsamin) [inline sdk/project.go]: godoc

File: ui/src/app/app.service.ts [hunk 8]
```diff
@@ -41,35 +43,43 @@ export class AppService {
             case 'pipeline':
                 this.updatePipelineCache(lastUpdate);
                 break;
+            case 'workflow':
+                this.updatePipelineCache(lastUpdate);
```
Comment (sguiheux) [inline ui/src/app/app.service.ts]: updateWorkflowCache?

File: ui/src/app/app.service.ts [hunk 9]
```diff
@@ -178,4 +188,43 @@ export class AppService {
             }
         });
     }
+
+    updateWorkflowCache(lastUpdate: LastModification): void {
+        this._wfStore.getWorkflows(lastUpdate.name).first().subscribe(wfs => {
```
Comment (sguiheux) [inline ui/src/app/app.service.ts]: .first => pipe ?

**Pull request merged**